lib/support/modules/PesterUtil/PesterUtil.psm1

```
@@ -0,0 +1,143 @@
+        $wc.Proxy = $webproxy
+    }
+
+    Write-Verbose -Object "Downloading Package from $nupkgUrl" 
+    $wc.DownloadFile($nupkgUrl,$downloadedZip)
+    if (-not (Test-Path -Path $downloadedZip)) {
+        Throw "Error trying to download nupkg '$nupkgUrl' to '$downloadedZip'."
+    }
+    
+    # Test to see if Expand-Archive is available first
+    if (Get-Command Expand-Archive) {
+        Expand-Archive -Path $downloadedZip -DestinationPath $ModuleFolder -Force
+    }
+    else {
+        # Fall back to COM object for Shell.Application zip extraction
+        Write-Host "Creating COM object for zip file '$downloadedZip'."
+        $shellcom = New-Object -ComObject Shell.Application
+        $zipcomobject = $shellcom.Namespace($downloadedZip)
+        $destination = $shellcom.Namespace($ModuleFolder)
+        $destination.CopyHere($zipcomobject.Items(), 0x610)
+        Write-Host "Nupkg installed at $ModuleFolder"
+    }
+
+    [GC]::Collect()
+}
+
+function Set-PSRepo {
+    param(
+        [Parameter(Mandatory)]
+        $Repository
+    )
+    if (-not (Get-Command Get-PSRepository) -and (Get-Command Get-PackageSource)) {
+        # Old version of PSGet do not have a *-PSrepository but have *-PackageSource instead.
+        if (Get-PackageSource -Name $Repository.Name)  {
+            Set-PackageSource @Repository
+        }
+        else {
+            Register-PackageSource @Repository
+        }
+    }
+    elseif (Get-Command Get-PSRepository) {
+        if (Get-PSRepository -Name $Repository.Name -ErrorAction SilentlyContinue) {
+            # The repo exists, we should use Set-PSRepository and splat parameters
+            Set-PSRepository @Repository
+        }
+        else {
+            # The repo does not exist, use Register-PSRepository and splat
+            Register-PSRepository @Repository
+        }
+    }
+    else {
+        Write-Host "Cannot Set PS Repository, command Set or Register for PSRepository or PackageSource not found."
+    }
```

Should we throw an error in this case?